mautic/mautic
# Block access to contacts everywhere

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [yes]
| New feature/enhancement? (use the a.x branch)      | [no]
| Deprecations?                          | [no]
| BC breaks? (use the c.x branch)        | [no]
| Automated tests included?              | [no] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:


This PR fixes the handling of unexpected exceptions and the access to contact information without the necessary permissions.

Before:
![image](https://user-images.githubusercontent.com/96085911/161594038-9a703457-0a9e-4e76-8e96-dc3c8113f25e.png)

After:
![image](https://user-images.githubusercontent.com/96085911/161593939-4b228dbc-124a-49e1-b172-554a2e1c02f5.png)

#### Steps to reproduce this issue:
1. Remove all contact permissions (email:emails:viewown,email:emails:viewother)
2. Navigate to a segment details view, click on contacts
3. The contacts included in the segment are visible

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Repeat Steps 1 and 2
3. A message that no contacts are available or that the permissions are missing is displayed.






<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11042"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
